### PR TITLE
Makes x-mas trees automatically spawn

### PR DESCRIPTION
**Pull request by lzimann**
:cl: Lzimann
add: The robots stole Santa's Elfs jobs.
/:cl:

Adds a x-mas tree/festivus pole landmark spawned, also adds the holiday event "Festive Season", which will start 1st december and end at 31. The event will handle tree spawning. 

Landmarks were added to all maps.

### CODE REVIEW

File: code/modules/holiday/holidays.dm [hunk 1]
```diff
@@ -316,6 +316,26 @@
...
+	end_day = 31
+
+/datum/holiday/festive_season/celebrate()
+	for(var/obj/effect/landmark/xmastree/XT in world)
+		var/turf/t = get_turf(XT)
+		new /obj/structure/flora/tree/pine/xmas(t)
+		qdel(XT)
+
+/datum/holiday/festive_season/greet()
+	return "Have a nice festive season!"
```
Comment (KorPhaeron) [inline code/modules/holiday/holidays.dm]: when will the war on christmas end?

File: code/modules/holiday/holidays.dm [hunk 2]
```diff
@@ -316,6 +316,26 @@
 /datum/holiday/xmas/greet()
 	return "Have a merry Christmas!"
 
+/datum/holiday/xmas/celebrate()
+	for(var/obj/effect/landmark/xmastree/XT in world)
```
Comment (KorPhaeron) [inline code/modules/holiday/holidays.dm]: Never in world

Landmarks are already a list iirc

File: code/modules/holiday/holidays.dm [hunk 3]
```diff
@@ -316,6 +316,26 @@
...
+		qdel(XT)
+
+/datum/holiday/festive_season
+	name = FESTIVE_SEASON
+	begin_day = 1
+	begin_month = DECEMBER
+	end_day = 31
+
+/datum/holiday/festive_season/celebrate()
+	for(var/obj/effect/landmark/xmastree/XT in world)
```
Comment (KorPhaeron) [inline code/modules/holiday/holidays.dm]: Same as above

File: code/modules/holiday/holidays.dm [hunk 4]
```diff
@@ -316,6 +316,26 @@
...
+		var/turf/t = get_turf(XT)
+		new /obj/structure/flora/tree/pine/xmas(t)
```
Comment (KorPhaeron) [inline code/modules/holiday/holidays.dm]: why not just new obj/blah(get_turf(XT))

Comment (lzimann) [inline code/modules/holiday/holidays.dm]: i did that but in the wrong holiday datum

**Pull request merged**